plugins/woocommerce/includes/wc-update-functions.php

```diff
@@ -2589,3 +2590,30 @@ function wc_update_770_remove_multichannel_marketing_feature_options() {
 	delete_option( 'woocommerce_multichannel_marketing_enabled' );
 	delete_option( 'woocommerce_marketing_overview_welcome_hidden' );
 }
+
+/**
+ * Delete posts of type "shop_order_placeholder" with no matching order in the orders table.
+ *
+ * @return void
+ */
+function wc_update_790_delete_stray_order_records() {
+	global $wpdb;
+
+	$orders_table_name = OrdersTableDataStore::get_orders_table_name();
+
+	// phpcs:disable WordPress.DB.PreparedSQL
+
+	if ( ! $wpdb->get_var( $wpdb->prepare( 'SHOW TABLES LIKE %s', $wpdb->esc_like( $orders_table_name ) ) ) ) {
```

Comment:
We would need to suppress errors here like so https://github.com/WordPress/wordpress-develop/blob/b98b5e59d72a68bc31f99b866fa26f7769d16e97/src/wp-admin/includes/upgrade.php#L2796-L2798 otherwise they will show in the $wpdb error log.

Reply:
Implemented in https://github.com/woocommerce/woocommerce/pull/37050/commits/ae1e340ae65da1d73e3727604e30c50e68351cff